lark/grammar.py

```diff
@@ -50,11 +52,11 @@ class RuleOptions(Serialize):
     __serialize_fields__ = 'keep_all_tokens', 'expand1', 'priority', 'template_source', 'empty_indices'
 
     def __init__(self, keep_all_tokens=False, expand1=False, priority=None, template_source=None, empty_indices=()):
-        self.keep_all_tokens = keep_all_tokens
-        self.expand1 = expand1
-        self.priority = priority
-        self.template_source = template_source
-        self.empty_indices = empty_indices
+        self.keep_all_tokens: bool = keep_all_tokens
```

Comment:
I would personally prefer keeping the types out of `__init__` and instead have them on the class level. But we can argue here.

Reply:
I'm against any duplication of information, as much as we can help it.

Reply:
I think it makes sense to declare all the instance variables on the class level and keep them in one place. I have noticed a few classes that initialize instance variables outside `__init__` which would make type annotations spread out.